src/core/csv/reader_parsers.cc

```diff
@@ -277,7 +284,7 @@ void parse_float32_hex(FreadTokenizer& ctx) {
       else if (E == 126 && Eneg && acc) /* subnormal */ E = 0;
       else goto fail;
     } else {
-      E = 127 + (E ^ -Eneg) + Eneg;
+      E = 127 + (E ^ static_cast<uint32_t>(-static_cast<int>(Eneg))) + Eneg;
```

Comment:
much clearer:
```suggestion
      E = Eneg? 127 - E : 127 + E;
```
or use `127u` if compiler still complains.

Reply:
Sure, I thought there was some reason for the original cryptic formulation?